### PR TITLE
xrootd/xrdproto/locate: implement kXR_locate

### PR DESCRIPTION
**Pull request by sbinet**
Updates go-hep/hep#170.

### CODE REVIEW

File: xrootd/xrdproto/locate/locate.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	w.WriteI32(int32(len(o.Path)))
+	w.WriteBytes([]byte(o.Path))
+	return nil
+}
+
+// UnmarshalXrd implements xrdproto.Unmarshaler.
+func (o *Request) UnmarshalXrd(r *xrdenc.RBuffer) error {
+	o.Options = r.ReadU16()
+	r.Skip(14)
+	n := r.ReadI32()
```
Comment (EgorMatirov) [inline xrootd/xrdproto/locate/locate.go]: Let's use `ReadStr` and `WriteStr` instead.

File: xrootd/xrdproto/locate/locate.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+func (req *Request) ReqID() uint16 { return RequestID }
+
+// ShouldSign implements xrdproto.Request.ShouldSign.
+func (*Request) ShouldSign() bool { return false }
+
+// MarshalXrd implements xrdproto.Marshaler.
+func (o Request) MarshalXrd(w *xrdenc.WBuffer) error {
+	w.WriteU16(o.Options)
+	w.Next(14)
+	w.WriteI32(int32(len(o.Path)))
```
Comment (EgorMatirov) [inline xrootd/xrdproto/locate/locate.go]: Let's use `ReadStr` and `WriteStr` instead.

Comment (sbinet) [reply xrootd/xrdproto/locate/locate.go]: right!

File: xrootd/xrdproto/locate/locate.go (hunk 1 above)
Comment (sbinet) [reply xrootd/xrdproto/locate/locate.go]: done.

**Pull request merged**